File: frontend/src/types/lesson-resource.ts
export type ResourceType =
  | "PDF"
  | "PowerPoint"
  | "Word"
  | "Text"
  | "Video"
  | "Audio"
  | "Image"
  | "ZIP"
  | "YouTube"
  | "GitHub"
  | "Website"
  | "Reference Book"
  | "Other";

export interface LessonResource {
  id: number;
  lesson_id: number;
  title: string;
  resource_type: ResourceType | string;
  url?: string | null;
  file_path?: string | null;
  file_size?: number | null;
  author?: string | null;
  description?: string | null;
  order_number?: number;
  visibility?: "Published" | "Draft" | "Hidden";
  created_at: string;
}

export interface LessonResourceCreateRequest {
  title: string;
  resource_type: string;
  file?: File | null;
  url?: string | null;
  author?: string | null;
  description?: string | null;
}

export interface LessonResourceUpdateRequest {
  title?: string;
  resource_type?: string;
  url?: string | null;
  author?: string | null;
  description?: string | null;
}

export interface LessonResourceGroupedResponse {
  pdfs: LessonResource[];
  videos: LessonResource[];
  youtube: LessonResource[];
  github: LessonResource[];
  ppt: LessonResource[];
  books: LessonResource[];
  notes: LessonResource[];
  links: LessonResource[];
  all_resources: LessonResource[];
}

File: backend/app/schemas/lesson_resource.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class LessonResourceCreate(BaseModel):
    title: str
    resource_type: str
    url: str | None = None
    file_path: str | None = None
    file_size: int | None = None
    author: str | None = None
    description: str | None = None


class LessonResourceUpdate(BaseModel):
    title: str | None = None
    resource_type: str | None = None
    url: str | None = None
    author: str | None = None
    description: str | None = None


class LessonResourceResponse(BaseModel):
    id: int
    lesson_id: int
    title: str
    resource_type: str
    url: str | None = None
    file_path: str | None = None
    file_size: int | None = None
    author: str | None = None
    description: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class LessonResourceGroupedResponse(BaseModel):
    pdfs: list[LessonResourceResponse] = []
    videos: list[LessonResourceResponse] = []
    youtube: list[LessonResourceResponse] = []
    github: list[LessonResourceResponse] = []
    ppt: list[LessonResourceResponse] = []
    books: list[LessonResourceResponse] = []
    notes: list[LessonResourceResponse] = []
    links: list[LessonResourceResponse] = []
    all_resources: list[LessonResourceResponse] = []
